### PR TITLE
More unit test cases to ensure LogSegment ordering

### PR DESCRIPTION
**Pull request by vgkholla**
Also adds the right comparator on maps in the stats classes

### CODE REVIEW

**Comment by vgkholla** [comment]
@nsivabalan  fixes for bugs - please review

File: ambry-store/src/main/java/com.github.ambry.store/BlobStoreStats.java [hunk 1]
```diff
@@ -378,7 +378,7 @@ public void close() {
         referenceTimeInMs);
     long startTimeMs = time.milliseconds();
     Map<StoreKey, Long> deletedKeys = new HashMap<>();
-    NavigableMap<String, Long> validSizePerLogSegment = new TreeMap<>();
+    NavigableMap<String, Long> validSizePerLogSegment = new TreeMap<>(LogSegmentNameHelper.COMPARATOR);
```
Comment (nsivabalan) [inline ambry-store/src/main/java/com.github.ambry.store/BlobStoreStats.java]: can you create an issue to add tests in BlobStoreStatsTest to test this scenario (I assume you haven't covered in this patch)

**Comment by vgkholla** [comment]
@nsivabalan you are the only reviewer, could you merge?

File: ambry-store/src/main/java/com.github.ambry.store/BlobStoreStats.java (hunk 1 above)
Comment (nsivabalan) [inline ambry-store/src/main/java/com.github.ambry.store/BlobStoreStats.java]: Can you give me the link to the issue ?

Comment (vgkholla) [inline ambry-store/src/main/java/com.github.ambry.store/BlobStoreStats.java]: I had created an issue and assigned it to you when the bug was found (#713). We can use the same one to track tests.

**Pull request merged**